v2/validator.go

```diff
@@ -0,0 +1,45 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package v2
+
+import (
+	"fmt"
+	"github.com/go-playground/validator/v10"
+)
+
+var val *validator.Validate
+
+func init() {
+	val = validator.New()
+}
+
+// Validate function will use the validator package to validate the struct annotation
+func Validate(a interface{}) error {
+	err := val.Struct(a)
+	// translate all error at once
+	if err != nil {
+		errs := err.(validator.ValidationErrors)
+		for _, e := range errs {
+			return getErrorMessage(e.Tag(), e.Field(), e.Param())
```

Comment:
If there are multiple validation errors, this will only return a message for the first error. Instead of returning, should build a a comma separated list of the errors and return the combined error message after the loop.

Reply:
Also, just pass `e` rather than the fields in `e` so the missing switch default case can depend on `e` for error message.